jaxopt/_src/lbfgs.py

```diff
@@ -256,19 +262,34 @@ def init_state(self,
     Returns:
       state
     """
+    if isinstance(init_params, base.OptStep):
+      # this allows to maintain backward compatibility with the previous API
+      hessian_approx_kwargs = dict(
+        s_history=init_params.state.s_history,
+        y_history=init_params.state.y_history,
+        rho_history=init_params.state.rho_history,
+        gamma=init_params.state.gamma,
+        iter_num=init_params.state.iter_num,
+      )
+      init_params = init_params.params
+      dtype = tree_single_dtype(init_params)
+    else:
+      dtype = tree_single_dtype(init_params)
+      hessian_approx_kwargs = dict(
+        s_history=init_history(init_params, self.history_size),
+        y_history=init_history(init_params, self.history_size),
+        rho_history=jnp.zeros(self.history_size, dtype=dtype),
+        gamma=jnp.asarray(1.0, dtype=dtype),
+        iter_num=jnp.asarray(0),
+      )
     if self.has_aux:
       _, aux = self.fun(init_params, *args, **kwargs)
     else:
       aux = None
-    dtype = tree_single_dtype(init_params)
-    return LbfgsState(iter_num=jnp.asarray(0),
-                      value=jnp.asarray(jnp.inf),
+    return LbfgsState(value=jnp.asarray(jnp.inf),
                       stepsize=jnp.asarray(self.max_stepsize, dtype=dtype),
                       error=jnp.asarray(jnp.inf),
```

Comment:
Shouldn't we initialize `value`, `stepsize` and `errror` too?

Reply:
I think it's a good question. 
For `value` and `error` there is not really any interest in doing so, it's not going to affect how the algorithm runs.
For stepsize it's a bit different since the linesearch is based off of that.
I would say it's probably best to give the user the choice so initialize `stepsize` with the value contained in the state, which the user can always change, and make sure it's well documented.